### DISCUSSION TITLE
I have several Entities showing as Unkown

### DISCUSSION THREAD

**Discussion by thorrrr** [BottlecapDave/HomeAssistant-OctopusEnergy#861] [Q&A]
Octopus Missing Entities

How can i fix these please 

sensor.octopus_energy_electricity_1Redacted74_Redacted_next_rate

sensor.octopus_energy_electricity_1Redacted74_Redacted_previous_rate

Gas

sensor.octopus_energy_gas_Redacted_Redacted_next_rate

sensor.octopus_energy_gas_Redacted_Redacted_previous_rate

**Comment by BottlecapDave** [comment]
Hello and sorry you're having issues. These sensors will only be unknown if the integration is failing to retrieve then latest rate information.

* What tariff are you on (full code available as attribute on sensor)?
* Do you have any errors in your logs?
* Do they come back if you reload the integration?
* Are you on the latest release?

**Comment by thorrrr** [comment]
HI Dave
On Octopus Flexible  
Latest Version
They  do not change after reboot 
I tried to download diagnostics  but does not work for me 
![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/8170673/6ec37921-44e3-4b4f-8242-19d605bd449b)





This error originated from a custom integration.

Logger: custom_components.octopus_energy.coordinators.current_consumption
Source: custom_components/octopus_energy/coordinators/current_consumption.py:55
integration: Octopus Energy (documentation, issues)
First occurred: 00:55:08 (2 occurrences)
Last logged: 00:55:37

Failed to retrieve smart meter consumption data - using cached version. Next attempt at 2024-05-02 00:54:48.293695+01:00
Failed to retrieve smart meter consumption data - using cached version. Next attempt at 2024-05-02 00:56:14.141937+01:00





This error originated from a custom integration.

Logger: custom_components.octopus_energy.api_client
Source: custom_components/octopus_energy/api_client/__init__.py:702
integration: Octopus Energy (documentation, issues)
First occurred: 00:55:08 (2 occurrences)
Last logged: 00:55:37

Failed to connect. Timeout of ClientTimeout(total=None, connect=None, sock_read=20, sock_connect=20, ceil_threshold=5) exceeded.

**Reply by BottlecapDave** [reply]
If you're getting client timeout issues, then there is very little I can do as this suggests a connection issue either with your HA instance or between you and the API.

However, a failure to retrieve current consumption data shouldn't effect rate based sensors.

The fact the download diagnostics doesn't work enforces this as it makes a call to get your account information. Has your API key changed? I also need the full tariff code (e.g. E-1R-xxx-A) so that I can look up the rates and make sure they're available (although I don't think this is your issue). Are you doing anything else with the API or have any other OE related integrations installed (and therefore getting rate limited)?

**Comment by Psai** [comment]
I wonder if this is the same issue as me...

A few days ago, a lot of entities became "unknown".
After rebooting, it says off permanently, although my tariff hasn't changed.
![Screenshot_20240502-152126~2](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/3990521/a528300a-a811-4f3b-a6a8-19bf2fe69ebb)

I also have lots of warning logs:
![Screenshot_20240502-152858](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/3990521/8bf6c5fc-17f4-4de1-9753-3975258d4d8a)

**Reply by BottlecapDave** [reply]
Like Thorr, your instance of HA looks like it's having issues connecting to the OE API which is something beyond my control. If the integration can't get the data, then it can't update the sensors. Have you also tried reloading the integration or your HA instance?

The instance in your log which is curious is where it states "Errors in request".

**Reply by thorrrr** [reply]
I have found today no Gas data coming through 
ALso diagnostics works from all other intergrations but not Octopus!
![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/8170673/c32ce54b-1705-45ec-92c3-dda8f1fd1ec7)
![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/8170673/71e176d2-4ba0-4959-b6a6-5b69f097d287)

**Reply by BottlecapDave** [reply]
Are you still having issues? Have you tried reconfiguring the api key like @Psai? Have you tried updating to the latest release?

**Reply by Psai** [reply]
Fyi, I was wrong. Offpeak/intelligent dispatching hasn't triggered since my first message. I've reconfigured multiple times Including removing via hacs and restarting the home assistant machine completely.
I reverted to just using a scheduled time to stop/start the automations.

**Reply by BottlecapDave** [reply]
@Psai have you tried upgrading to the latest version. Based on some of the warnings it looks like you're on an old version of the integration.

**Comment by Psai** [comment]
Verrrryyy interesting.
I refreshed the API key, and it looks like it *might* now be working again
(although the previous API key was correct)

**Comment by Psai** [comment]
I'm on the v11.0.0

**Reply by BottlecapDave** [reply]
I only asked as your screenshot referenced "EntityType is used..." which was fixed in v11. Are you getting similar warnings of not being able to connect?
